### DISCUSSION TITLE
Projects using log4j 2.12.1 are affected by CVE-2021-44228, but are not detected yet.

### DISCUSSION THREAD

**Discussion by qianwch** [DependencyTrack/dependency-track#1300] [General]
It is clear that my projects are using log4j 2.12.1, but dependency track could not detect VULs about CVE-2021-44228. What could we do to solve this?

**Comment by stevespringett** [comment]
OSS Index is not properly identifying the vulnerability on 2.12.2, but is identifying it on 2.12.1.
https://ossindex.sonatype.org/component/pkg:maven/org.apache.logging.log4j/log4j-core@2.12.2
https://ossindex.sonatype.org/component/pkg:maven/org.apache.logging.log4j/log4j-core@2.12.1

Please report these issues to OSS Index. Once resolved, DT will correctly identify it.
https://ossindex.sonatype.org/doc/report-vulnerability

**Comment by qianwch** [comment]
It is version 2.12.1，I just mistyped..
<img width="1102" alt="截屏2021-12-15 上午10 59 40" src="https://user-images.githubusercontent.com/45196316/146115349-941e2659-8883-491d-8db5-f20ce0ef7f8c.png">
.

**Reply by stevespringett** [reply]
log4j-api is not vulnerable. It provides an API contract, but no code implementations. log4j-to-slf4j is a bridge between two logging frameworks. To my knowledge, log4j-core is the only module that is vulnerable.

**Comment by officerNordberg** [comment]
This looks fine to me, the vuln is in log4j-core, if logs are being bridged
to slf4j who can say what the implementation is.  Correctly not reported.

**Comment by savujevi** [comment]
I have uploaded a sbom to dependencytrack including pkg:maven/org.apache.logging.log4j/log4j-core@2.13.3.

<img width="1184" alt="Screenshot 2021-12-16 at 10 33 37" src="https://user-images.githubusercontent.com/644396/146345743-bb70cf35-4c9e-4e41-b5b5-f2c32b715b2a.png">

<img width="576" alt="Screenshot 2021-12-16 at 10 38 47" src="https://user-images.githubusercontent.com/644396/146346649-6ef34646-6ac5-425b-b4a5-3c1ef81b5363.png">

There is no vulnerability found.

It looks like the identifier is present:
https://ossindex.sonatype.org/component/pkg:maven/org.apache.logging.log4j/log4j-core@2.13.3

Sonatype OSS Index is on.

<img width="749" alt="Screenshot 2021-12-16 at 10 35 04" src="https://user-images.githubusercontent.com/644396/146345982-fe31615c-aaf6-4377-a19c-ed1b026a76bb.png">

I see nothing special in the logfile:

`10:17:00.876 INFO [OssIndexAnalysisTask] Starting Sonatype OSS Index analysis task
10:17:04.945 INFO [OssIndexAnalysisTask] Sonatype OSS Index analysis complete
10:17:04.947 INFO [PolicyEngine] Evaluating 522 component(s) against applicable policies
10:17:06.353 INFO [PolicyEngine] Policy analysis complete
10:17:06.358 INFO [MetricsUpdateTask] Executing metrics update for project: 619c700e-9330-4c9f-9ac1-22d1517203e7
10:17:17.150 INFO [MetricsUpdateTask] Completed metrics update for project: 619c700e-9330-4c9f-9ac1-22d1517203e7`

Generely vulnerabilities are found:
<img width="1215" alt="Screenshot 2021-12-16 at 10 36 58" src="https://user-images.githubusercontent.com/644396/146346379-9612373f-1a38-4eae-809f-ae6441303d62.png">

Do I miss something ?

Thank you for your help.

**Reply by stevespringett** [reply]
The vulnerability could also be identified as `f0ac54b6-9b81-45bb-99a4-e6cb54749f9d`

**Comment by savujevi** [comment]
Does anyone have checked a log4j-core 2.x in dependency-track only to check if my installation is ok or not.
Thanks.

**Reply by stevespringett** [reply]
Dependency-Track does not use log4j

**Reply by savujevi** [reply]
I have expressed myself in a misleading way.

I mean that I have uploaded a sbom from an apllication of us in dependency-track with pkg:maven/org.apache.logging.log4j/log4j-core@2.13.3 included, but this log4j-core finding does not appear in the audit vulnerabilities as you can see in the pictures.

The question is if I have installed something wrong according to the images I have posted.

If I follow the link https://ossindex.sonatype.org/component/pkg:maven/org.apache.logging.log4j/log4j-core@2.13.3 I get a result, but this result is not present in dependency-track in my project.

I am not asking if dependency-track uses log4j.

Thank you for your help.

**Comment by savujevi** [comment]
Now the vulnerability for log4j is shown. Strange. Maybe a timing problem or just a restart ..
Sorry for the noise and thank you for your help and the great work.
